### PR TITLE
Use magic number filter when checking positions

### PR DESCRIPTION
**Pull request by yzAlvin**
- OnTick now only modifies positions after checking positions total by `Magic`
- CloseAllPositions also just continues if the position is not `Magic` or `Symbol`

nice change cause otherwise it will modify all open positions, even those initiated by other EAs

### CODE REVIEW

File: Strategies/Box Breakout/Box Breakout v1.1.mq5 [hunk 1]
```diff
@@ -388,6 +395,14 @@ void SetTrailingSL(double ask, double bid, ulong posTicket) {
 //+------------------------------------------------------------------+
 //| Utilities                                                        |
 //+------------------------------------------------------------------+
+int PositionsTotalByMagic(int magic) {
+  int total = 0;
+  for (int i=0; i < PositionsTotal(); i++) {
+    if (PositionGetInteger(POSITION_MAGIC) == magic) total++;
+  }
+  return total;
+}
```
Comment (yzAlvin) [inline Strategies/Box Breakout/Box Breakout v1.1.mq5]: This really feels like it could be simplified..

**Comment by je-zhou** [comment]
Hey Alvin, thanks for your suggestion. Totally agree with this change, I only run this single EA so haven't needed to handle positions entered into by other EAs. This is a good idea to expand your portfolio to run multiple strategies at a time. 

There's a small bug in your code above I was hoping you could fix. Your PositionsTotalByMagic function is referencing the same Position for the duration of the for loop, but I am assuming you want to count only the positions for a given magic number? You need to call PositionGetTicket first to preload the position before you call PostionGetInteger.

See docs @ https://www.mql5.com/en/docs/trading/positiongetticket

```
int PositionsTotalByMagic(int magic) {
  int total = 0;
  for (int i=0; i < PositionsTotal(); i++) {
    PositionGetTicket(i) //--- You need to preload this ticket before you can call PositionGetInteger
    if (PositionGetInteger(POSITION_MAGIC) == magic) total++;
  }
  return total;
}
```

Once you make the change above, happy to merge this! 👍

**Comment by yzAlvin** [comment]
Seems weird to call `PositionGetTicket(i)`, which returns a `ulong`, but then we don't save it to a variable to use it later,

nevertheless, i made the change

**Comment by je-zhou** [comment]
Thanks! The `ulong` being returned is the Position Identifier which can be used for logging purposes. Because the purpose of your `PositionsTotalByMagic` function is to just return a count of positions, I don't think logging position ids here would be super informative.

**Pull request merged**